src/app/courses/courses.component.ts

```diff
@@ -1,30 +1,66 @@
 import { Component, OnInit } from '@angular/core';
 
 import { ICourse } from './shared/course.model';
-import { FilterByTitlePipe } from './filter-by-title.pipe';
 import { CoursesService } from './courses.service';
 import { DialogService } from '../shared/dialog/dialog.service';
+import { LoaderService } from '../shared/loader/loader.service';
+import { ICoursesResponse } from './shared/courses-response.model';
 
 @Component({
   selector: 'courses',
   templateUrl: './courses.component.html',
   styleUrls: ['./courses.component.scss'],
 })
 export class CoursesComponent implements OnInit {
+  public isLoading: boolean;
   public courses: ICourse[] = [];
+  private page = 1;
+  private count: number;
 
   constructor(
     private coursesService: CoursesService,
-    private filterByTitle: FilterByTitlePipe,
-    private dialogService: DialogService
-  ) {}
+    private dialogService: DialogService,
+    private loaderService: LoaderService
+  ) {
+    loaderService.isLoading.subscribe(
```

Comment:
Please, check all of your subscriptions above and bellow and fix the memory leak.